### PR TITLE
feat(sdk): Add a component decorator for v2 components.

### PR DESCRIPTION
**Pull request by neuromage**
**Checklist:**
- [X] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 
- [ ] Do you want this pull request (PR) cherry-picked into the current release branch?

### CODE REVIEW

File: sdk/python/kfp/v2/dsl/__init__.py [hunk 1]
```diff
@@ -12,4 +12,13 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from kfp.dsl import *
```
Comment (chensun) [inline sdk/python/kfp/v2/dsl/__init__.py:15]: Per the presubmit test results, some referencing of `from kfp.v2 import dsl` needs to be fixed.

File: sdk/python/kfp/v2/dsl/__init__.py [hunk 2]
```diff
@@ -12,4 +12,13 @@
...
+from kfp.v2.dsl.component import component
```
Comment (Ark-kun) [inline sdk/python/kfp/v2/dsl/__init__.py:15]: This looks like a circular reference. `kfp.v2.dsl` is importing from `kfp.v2.dsl`.

Perhaps you can use `from .component import component`

File: sdk/python/kfp/v2/dsl/component.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
+# Copyright 2021 Google LLC
```
Comment (Ark-kun) [inline sdk/python/kfp/v2/dsl/component.py:1]: Do we want to leave this whole module file public?
This results in the following imports being available:
* `kfp.v2.dsl.component`
* `kfp.v2.dsl.component.component`
* `kfp.v2.dsl.component.components`

**Comment by Ark-kun** [comment]
Other than the circular reference and test failure,
/lgtm

File: sdk/python/kfp/v2/dsl/component.py (hunk 3 above)
Comment (neuromage) [reply sdk/python/kfp/v2/dsl/component.py:1]: No, the last two imports are not available.

File: sdk/python/kfp/v2/dsl/__init__.py (hunk 2 above)
Comment (neuromage) [reply sdk/python/kfp/v2/dsl/__init__.py:15]: I don't think there's a circular reference here.

File: sdk/python/kfp/v2/dsl/__init__.py (hunk 1 above)
Comment (neuromage) [reply sdk/python/kfp/v2/dsl/__init__.py:15]: Done.

File: sdk/python/kfp/v2/dsl/__init__.py (hunk 2 above)
Comment (neuromage) [reply sdk/python/kfp/v2/dsl/__init__.py:15]: Sorry, I was wrong, fixed the filename.

**Comment by chensun** [comment]
/lgtm
/approve

Thanks!

**Pull request merged**